### PR TITLE
Add resource-management subproject to sig-node

### PR DESCRIPTION
**Pull request by klueska**
Add resource-management subproject to sig-node

As part of this, add `dynamic-resource-management` staging repo as part of this subproject as well as absorb the (previously) standalone subproject for `noderesourcetopology-api` into this new subproject.

/cc @derekwaynecarr 
/cc @swatisehgal 
/cc @pohly

### CODE REVIEW

**Comment by pohly** [comment]
/assign @palnabarun

For approval.

File: sigs.yaml [hunk 1]
```diff
@@ -2148,8 +2148,9 @@ sigs:
     owners:
     - https://raw.githubusercontent.com/kubernetes/node-problem-detector/master/OWNERS
     - https://raw.githubusercontent.com/kubernetes/test-infra/master/config/jobs/kubernetes/node-problem-detector/OWNERS
-  - name: noderesourcetopology-api
+  - name: resource-management
     owners:
+    - https://raw.githubusercontent.com/kubernetes/dynamic-resource-allocation/master/OWNERS
```
Comment (mrbobbytables) [inline sigs.yaml:2153]:
```suggestion
    - https://raw.githubusercontent.com/kubernetes/kubernetes/master/staging/src/k8s.io/dynamic-resource-allocation/OWNERS
```

For staging repos, they should point to their location in k/k as that is the authoritative place for the code 👍

Comment (klueska) [reply sigs.yaml]: Updated thanks.

**Review by mrbobbytables — COMMENTED** [review]
/lgtm
/approve

**Comment by derekwaynecarr** [comment]
just caught up on slack after holiday, happy to see this growth in the sig.

/approve

**Pull request merged**